// paymentwall/lib/paymentwall/onetimetoken.js
var Base = require('./base'),
	Product = require('./product'),
	util = require('util'),
	crypto = require('crypto');
	querystring = require('querystring');
	http = require('http');

function Onetimetoken(card_number,card_exp_month,card_exp_year,card_cvv){
	this.card={
		number: card_number || null,
		card_exp_month: card_exp_month || null,
		card_exp_year: card_exp_year || null,
		card_cvv: card_cvv || null
	}
	this.url = 'pwgateway.com';
	this.path = '/api/token';
	this.reponse = null;
	this.reg = /_+/;
	this.publickey = this.getAppKey();
	this.secretkey = this.getSecretKey();
}

util.inherits(Onetimetoken, Base);

util._extend(Onetimetoken.prototype, {

	getonetimetoken: function(){
	    var prefix_result = this.reg.test(this.secretkey); 

	    //set the post data
		var post_data = {
			'public_key': this.publickey,
			'card': {
				'number': this.card.number,
				'card_exp_month': this.card.card_exp_month,
				'card_exp_year': this.card.card_exp_year,
				'card_cvv': this.card.card_cvv
			}
	  	}

		post_data = querystring.stringify(post_data);
		console.log(post_data);

		//set the URL and path
		if (prefix_result==true) {
			this.url = 'api.paymentwall.com';
			this.path = '/api/brick/token';
		};
		//set the request options
		var post_options = {
		    host: this.url,
		    port: '80',
		    path: this.path,
		    method: 'Post',
		    headers: {
		        'Content-Type': 'application/x-www-form-urlencoded',
		        "X-ApiKey": this.secretkey
		    }
		};


		var post_req = http.request(post_options, function(error, response, body){	
	    if (error) {
	    	console.log(error);
	    	this.response = error;
	    };
	    resa.setEncoding('utf8');
	    resa.on('data', function (chunk) {
	        console.log('Response: ' + chunk);
	        this.response = body;
	    });
	    return this.response;
		});

		post_req.write(post_data);
		post_req.end();
	}
});
module.exports = Onetimetoken;
